test/reduce.js

```diff
@@ -274,19 +279,12 @@ module.exports = function reduce_test(testcase, minify_options, reduce_options)
 
                 // replace or skip statement
                 if (node instanceof U.AST_Statement) {
-                    if (node instanceof U.AST_BlockStatement
-                        && ( parent instanceof U.AST_IterationStatement
-                            || parent instanceof U.AST_Switch)) {
```

Comment:
This block is unreachable because:

#### `in_list && node instanceof AST_BlockStatement && parent instanceof AST_Switch`
`AST_Switch.body` is the only one that is `in_list`, and it can only contain the following node types:
- `AST_Case`
- `AST_Default`
both of which are not `AST_BlockStatement`.

#### `in_list && node instanceof AST_BlockStatement && parent instanceof AST_IterationStatement`
- `AST_IterationStatement.body` is not `in_list`
- None of the descendents of `AST_IterationStatement` contains `in_list` properties:
  - `AST_Do.condition`
  - `AST_While.condition`
  - `AST_For.init`
  - `AST_For.condition`
  - `AST_For.step`
  - `AST_ForIn.init`
  - `AST_ForIn.object`

Reply:
I'll take your word for it. Adding the in_list to the `if` condition was a late addition. I knew it used to crash when i hoisted AST_BlockStatement bodies from loops.